runtime/src/main/kotlin/kotlin/native/concurrent/Continuation.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.concurrent
+
+import kotlin.native.internal.*
+import kotlinx.cinterop.*
+
+public class Continuation0(block: () -> Unit,
+                    private val invoker: CPointer<CFunction<(COpaquePointer?) -> Unit>>,
+                    private val singleShot: Boolean = false): Function0<Unit> {
+
+    private var stable = StableRef.create(block)
```

Comment:
AFAIR, `StableRef` has direct object reference to the object, so `block` will be frozen too.

Reply:
Oh, that's unfortunate, and maybe shall be changed in StableRef impl. WDYT?

Reply:
I totally agree. `StableRef` shall be reimplemented as `inline class` over a pointer type. So it would become shareable by default.